language/move-prover/spec-lang/src/ast.rs

```diff
@@ -621,3 +638,74 @@ impl<'a> fmt::Display for ExpDisplay<'a> {
         Ok(())
     }
 }
+
+// =================================================================================================
+// # AST Expression Helpers
+
+impl Exp {
+    fn new_node(module_env: &ModuleEnv<'_>, typ: Type) -> NodeId {
+        module_env.new_node(None, typ)
```

Comment:
Each node also needs a location. If you absolutely don't have one, there is a GlobalEnv::unknown_loc